File: examples/basic-usage.py
"""
Basic usage example of PLP client
"""

import os
from plp_client import PLPClient, PromptInput

def main():
    # Initialize client
    client = PLPClient(
        "https://prompts.goreal.ai",
        api_key=os.getenv("PLP_API_KEY")  # Optional
    )

    try:
        # Create a new prompt
        print("Creating prompt...")
        client.put("examples/greeting", PromptInput(
            content="Hello {{name}}, welcome to {{product}}!",
            meta={
                "version": "1.0.0",
                "author": "demo",
                "description": "A simple greeting prompt"
            }
        ))

        # Retrieve the prompt
        print("Retrieving prompt...")
        prompt = client.get("examples/greeting")
        print(f"Prompt content: {prompt.content}")
        print(f"Prompt version: {prompt.meta.get('version')}")

        # Use the prompt with variables
        message = prompt.content\
            .replace("{{name}}", "Alice")\
            .replace("{{product}}", "PLP")
        print(f"Final message: {message}")

        # Clean up
        print("Deleting prompt...")
        client.delete("examples/greeting")

        print("✅ Done!")

    except Exception as e:
        print(f"Error: {e}")
        exit(1)

if __name__ == "__main__":
    main()

File: examples/basic-usage.js
/**
 * Basic usage example of PLP client
 */

const { PLPClient } = require('@plp/client');

async function main() {
  // Initialize client
  const client = new PLPClient('https://prompts.goreal.ai', {
    apiKey: process.env.PLP_API_KEY, // Optional
  });

  try {
    // Create a new prompt
    console.log('Creating prompt...');
    await client.put('examples/greeting', {
      content: 'Hello {{name}}, welcome to {{product}}!',
      meta: {
        version: '1.0.0',
        author: 'demo',
        description: 'A simple greeting prompt',
      },
    });

    // Retrieve the prompt
    console.log('Retrieving prompt...');
    const prompt = await client.get('examples/greeting');
    console.log('Prompt content:', prompt.content);
    console.log('Prompt version:', prompt.meta.version);

    // Use the prompt with variables
    const message = prompt.content
      .replace('{{name}}', 'Alice')
      .replace('{{product}}', 'PLP');
    console.log('Final message:', message);

    // Clean up
    console.log('Deleting prompt...');
    await client.delete('examples/greeting');

    console.log('✅ Done!');
  } catch (error) {
    console.error('Error:', error.message);
    process.exit(1);
  }
}

main();
